### ISSUE TITLE
Move file history to more options

### ISSUE THREAD

**Issue by clarkepaul** [silverstripe/silverstripe-asset-admin#581]
File history is being hidden as part of this issue https://github.com/silverstripe/silverstripe-asset-admin/issues/541. This issue is to make it visible again but in a new location which is in the more options dropdown for the file.

**Comment by chillu** [comment]
@clarkepaul I'm thinking of moving this out of the milestone, into a 4.1. What do you think the impact is for the average CMS author? I don't expect version tracking on files to be of very high value itself (e.g. knowing who changed a title, or replaced an image). On the other hand, it surfaces a banner feature of 4.x (filesystem abstraction), beyond just showing draft/live stages.

**Comment by clarkepaul** [comment]
@chillu Yeah, we actually came to the same conclusion when you were away. As part of this Beta we actually removed file history from view as it was still slowing down the edit panel load times and it didn't have the functionality to restore anyway. I'm happy to re-introduce the file history in the new location at a later date.

**Comment by chillu** [comment]
Discussed with Paul, draft files is a new feature and hence won't be impacting a lot of authors (just the subset that cares about individual publication of files). Chris Joe said that the current history panel implementation can be reactivated by config.

Deferring this will also give us an opportunity to look at history viewers in a wider context (content blocks etc)

**Comment by madmatt** [comment]
Is there any traction on this issue, or any plan for when it might be implemented? I've spoken to a couple of clients who are keen to use it so they can version files and preserve them for regulatory compliance.

Thanks!

**Comment by madmatt** [comment]
Also - is this the same as #873?

**Comment by robbieaverill** [comment]
It sounds like it is, will leave for @clarkepaul to put two cents in before closing one of them

**Comment by clarkepaul** [comment]
FYI I've closed the duplicate issue.

**Comment by eehondas** [comment]
Hi, I'm dev for one of the clients that spoke with @madmatt. Thanks Matt for pointing me to this issue. I can confirm we're really looking forward to this feature as we work with finance and they are pretty strict about compliances and being able to roll-back files. For what I can tell it's pretty close to be functional so here's hoping that you guys can fast track it. Also, on a similar note: another feature we are really looking forward to is the history tab for the blocks/elements to be extended to all DataObjects (after all an Element is a DataObject and looking at the tables on the database it looks like that view should be compatible?). If you guys can point me to a correct thread I'll officially request this second thing as well. Thanks!

**Comment by clarkepaul** [comment]
Thanks for the background @eehondas. The new history viewer has been built to be compatible with all types of data/files so its more of a matter for us to get the other UI's within the CMS to use it now (although there are a lot of other priorities).

**Comment by ScopeyNZ** [comment]
> its more of a matter [...] to get the other UI's within the CMS to use it now.

This can also be implemented with custom code (and custom DataObjects). There's some documentation here: https://docs.silverstripe.org/en/4/developer_guides/model/versioning/#using-the-history-viewer
